src/main/java/com/actiontech/dble/backend/datasource/PhysicalDbInstance.java

```diff
@@ -393,15 +394,20 @@ private void startHeartbeat() {
         }
 
         heartbeat.start();
-        heartbeat.setScheduledFuture(Scheduler.getInstance().getScheduledExecutor().scheduleAtFixedRate(() -> {
-            if (DbleServer.getInstance().getConfig().isFullyConfigured()) {
-                if (TimeUtil.currentTimeMillis() < heartbeatRecoveryTime) {
-                    return;
-                }
+        if (initHeartbeat.compareAndSet(false, true)) {
```

Comment:
why initHeartbeat doesn't need set false?

Reply:
It needs to be processed at stop